pkg/rbac/rbac.go

```diff
@@ -0,0 +1,314 @@
+		// [managedclusters] [devcluster1 devcluster2]}
+		// filter the rules by the given ApiGroup(or *) and Resource(or *))
+		ruleMatchesAPIGroup := (slices.Contains(rule.APIGroups, gr.Group) || slices.Contains(rule.APIGroups, "*"))
+		ruleMatchesResource := (slices.Contains(rule.Resources, gr.Resource) || slices.Contains(rule.Resources, "*"))
+
+		if !(ruleMatchesAPIGroup && ruleMatchesResource) {
+			continue
+		}
+
+		klog.V(2).Infof("Found Rule that matches the given GroupResource %v", rule)
+
+		// if a set of resource names are included in the rule, then add the acls only for those resources names
+		if len(rule.ResourceNames) != 0 {
+			for _, ruleResourceName := range rule.ResourceNames {
+				// if given resourcenames is empty or contains the resourcename in the rule,
+				//  apply rule to that resourcename
+				if len(resourcenames) == 0 || slices.Contains(resourcenames, ruleResourceName) {
+					// add verbs that are not already in the list
+					resourceAccessResults[ruleResourceName] = addUniqueItems(
+						resourceAccessResults[ruleResourceName], rule.Verbs...)
+				}
+			}
+		} else {
+			// if no resource names are set on the rule, it means acls in it apply to all resources of this type
+			if len(resourcenames) != 0 {
+				// if given resourcenames is nonempty,
+				// add acls to  each of the resourcename as the rule appplies to all  of the type
+				for _, rname := range resourcenames {
+					// add verbs that are not already in the list
+					resourceAccessResults[rname] = addUniqueItems(resourceAccessResults[rname], rule.Verbs...)
+				}
+			} else {
+				// if given resourcenames is empty,  add acls under the "*" entry as rule applied to all
+				resourceAccessResults["*"] = addUniqueItems(resourceAccessResults["*"], rule.Verbs...)
+			}
+		}
+	}
+
+	// the above only adds entries for the resourcenames that have some acls associated with it
+	// so add empty entries for any resourcenames that are missing in the result list
+	if len(resourcenames) != 0 {
+		for _, rname := range resourcenames {
+			if _, ok := resourceAccessResults[rname]; !ok {
+				resourceAccessResults[rname] = []string{}
+			}
+		}
+	}
+
+	klog.V(2).Infof("Resource access results %v", resourceAccessResults)
+
+	return resourceAccessResults, nil
+}
+
+// addUniqueItems a convenience method for building a list with unique entries
+// specified items are added to the given slice of items if not already in it
+func addUniqueItems(itemlist []string, itemsToAdd ...string) []string {
+	klog.V(2).Infof("Adding items %v to list %v", itemsToAdd, itemlist)
+
+	// iterate through each of the items in  itemsToAdd list
+	// and add item only if it doesnt already exist in the itemlist
+	resultList := itemlist
+	for _, item := range itemsToAdd {
+		// check if itemList  contains the item before appending
+		if !slices.Contains(resultList, item) {
+			resultList = append(resultList, item)
+		}
+	}
+
+	klog.V(2).Infof("Result List: %v ", resultList)
```

Comment:
This is likely too noisy. I'd even consider removing this log message.
```suggestion
	klog.V(3).Infof("Result List: %v ", resultList)
```